mirai-core-utils/src/jvmBaseMain/kotlin/Services.kt

```diff
@@ -13,11 +13,34 @@ import java.util.*
 import kotlin.reflect.KClass
 import kotlin.reflect.full.createInstance
 
+private enum class LoaderType {
+    JDK,
+    FALLBACK,
+}
+
+private val loaderType = when (systemProp("mirai.service.loader", "jdk")) {
+    "jdk" -> LoaderType.JDK
+    "fallback" -> LoaderType.FALLBACK
+    else -> throw IllegalStateException("mirai.service.loader must be jdk or fallback, cannot find a service loader")
+}
+
+private fun <T : Any> getJDKServices(clazz: KClass<out T>): ServiceLoader<out T> = ServiceLoader.load(clazz.java)
```

Comment:
```suggestion
private fun <T : Any> getJdkServices(clazz: KClass<out T>): ServiceLoader<out T> = ServiceLoader.load(clazz.java)
```
Kotlin 命名习惯是两字母缩写就用大写；三字母及以上首字母大写，其他字符小写。
IDEA 实现的一些 intention 是支持这个习惯的，例如把 `fun getID()` 转换为 `val`，IDE 会给出 `val id` 
`fun getJDK()` 则会变成 `val jdK` （我以前遇到过这种问题，但现在写的示例可能不准确）